packages/fx-core/src/component/resource/botService/appStudio/appStudioClient.ts

```diff
@@ -89,32 +99,33 @@ export class AppStudioClient {
     context?: ResourceContextV3
   ): Promise<void> {
     const axiosInstance = AppStudioClient.newAxiosInstance(token);
-    let create = false;
-    let response = undefined;
-    try {
-      if (registration.botId) {
-        const botReg = await AppStudioClient.getBotRegistration(token, registration.botId);
-        if (botReg) {
-          context?.logProvider?.info(Messages.BotResourceExist("Appstudio"));
-          return;
-        }
+
+    if (registration.botId) {
+      const botReg = await AppStudioClient.getBotRegistration(token, registration.botId);
+      if (botReg) {
+        context?.logProvider?.info(Messages.BotResourceExist("Appstudio"));
+        return;
       }
+    }
 
-      create = true;
-      response = await RetryHandler.Retry(() =>
+    try {
+      const response = await RetryHandler.Retry(() =>
         axiosInstance.post(`${AppStudioClient.baseUrl}/api/botframework`, registration)
       );
+      if (!isHappyResponse(response)) {
+        throw new ProvisionError(CommonStrings.APP_STUDIO_BOT_REGISTRATION);
+      }
     } catch (e) {
-      e.teamsfxUrlName = "<create-bot-registration>";
-      if (create) {
-        // Handle exception when creating bot failed.
-        throw new FailedToCreateBotRegistrationError(e.innerError);
+      if (e.response?.status === HttpStatusCode.UNAUTHORIZED) {
```

Comment:
The code here is a duplicate with the following, try to create a function to handle this?

Reply:
Somehow I prefer to make them explicit here for extensibility - We may add more status code handling in the future and status codes may vary between the two APIs.

Reply:
I have the same feeling here. Since we create dedicated Error Type for each error code, I think get/create/update API will share the same error handling. Maybe some error code doesn't exist in xxx API, but that doesn't impact the whole handling logic.

```
private static handleBotFrameworkError(e: Error, urlName: string, apiName: string): void {
  if (e.response?.status === HttpStatusCode.UNAUTHORIZED) {
    throw new BotFrameworkNotAllowedToAcquireTokenError();
  } else if (e.response?.status === HttpStatusCode.FORBIDDEN) {
    throw new BotFrameworkForbiddenResultError();
  } else if (e.response?.status === HttpStatusCode.TOOMANYREQS) {
    throw new BotFrameworkConflictResultError();
  } else {
    e.teamsfxUrlName = urlName;
    throw AppStudio.wrapException(e, apiName) as SystemError;
  }
```

Then get/create/update API can be simplified as
```
// Get API
} catch (e) {
  if (e.response?.status === HttpStatusCode.NOTFOUND) {
        return undefined; // Stands for NotFound.
  } else {
    handleBotFrameworkError(e, TeamsFxUrlNames.getBot, APP_STUDIO_API_NAMES.GET_BOT);
  }
}

// Create API
} catch (e) {
  handleBotFrameworkError(e, TeamsFxUrlNames.createBot, APP_STUDIO_API_NAMES.CREATE_BOT);
}

// Update API
} catch (e) {
  handleBotFrameworkError(e, TeamsFxUrlNames.updateEndpoint, APP_STUDIO_API_NAMES.UPDATE_BOT);
}
```

Reply:
Got it. I still prefer to leave it as it is for the same reason. As both of you have this suggestion, I decide to accept this and update it in another PR so this one can be merged soon.